packages/pages/src/dev/server/middleware/serverRenderRoute.ts

```diff
@@ -47,24 +48,36 @@ export const serverRenderRoute =
       const React = await import("react");
       const ReactDOMServer = await import("react-dom/server");
 
-      const { template, Component, props }: PageLoaderResult = await pageLoader(
-        {
-          url: url.pathname,
-          vite,
-          templateFilename: templateModuleInternal.filename,
-          entityId,
-          locale,
-          featuresConfig,
-          dynamicGenerateData,
-          projectStructure,
-        }
-      );
+      const { template, props }: PageLoaderResult = await pageLoader({
+        url: url.pathname,
+        vite,
+        templateFilename: templateModuleInternal.filename,
+        entityId,
+        locale,
+        featuresConfig,
+        dynamicGenerateData,
+        projectStructure,
+      });
+
+      if (templateModuleInternal.render) {
+        res
+          .status(200)
+          .set({ "Content-Type": "text/html" })
```

Comment:
maybe we use https://expressjs.com/en/api.html#res.type and get the extension from the path?

Falling back to text/plain otherwise.